elytron/ELY-2544-jaas-realm-via-custom-realm.adoc

```diff
@@ -0,0 +1,85 @@
+= WFLY-18163 Make it possible to use JaasSecurityRealm via a custom-realm resource
+:author:            Diana Krepinska
+:email:             dvilkola@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+This feature is about adding a possibility to add a JAAS based security realm via the custom-realm resource. Primary need for this feature is to allow users to use jaas-realm resource conveniently in older versions where it was not available and users want to use JDK 17.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-18163[WFLY-18163] community docs will be added here
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/ELY-2544[ELY-2544] - implementation
+* https://issues.redhat.com/browse/EAP7-2068[EAP7-2068] - EAP issue tracker
+* https://issues.redhat.com/browse/WFCORE-6411[WFCORE-6411] - Wildfly core custom-realm tests issue tracker
+
+=== Dev Contacts
+
+* mailto:dvilkola@redhat.com[Diana Krepinska]
+
+=== QE Contacts
+
+* mailto:TODO
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+[x] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly-security/wildfly-elytron[WildFly Elytron project]
+* https://github.com/wildfly-core/wildfly-core will have tests of custom-realm resource added
+* https://github.com/wildfly/wildfly will have community documentation added
+
+== Requirements
+
+=== Hard Requirements
+
+* It must be possible to add the `custom-realm` resource in the management model that will use the code of jaas-realm without an admin having to add any new classes as a module to the WildFly server.
+
+=== Nice-to-Have Requirements
+
+N/A
+
+=== Non-Requirements
+
+N/A
+
+== Implementation Plan
+
+The task will be achieved by adding custom realm wrapper around existing JaasSecurityRealm class. This custom realm wrapper will be deprecated as users should use jaas realm directly when available. This wrapper will be added to the `wildfly-core` repository and can be utilized by accessing a module `modules/system/layers/base/org/wildfly/extension/elytron`.
+
+Example of usage:
+
+```
+[standalone@localhost:9990 /] module add --name=jaasLoginModule --resources=path/to/login/modules/jar --dependencies=org.wildfly.security.elytron
+[standalone@localhost:9990 /] module add --name=elytronExtensionModule --dependencies=org.wildfly.extension.elytron,org.wildfly.security.elytron --resources=modules/system/layers/base/org/wildfly/extension/elytron
```

Comment:
Why is this needed? Why do we let this step up to users?

Reply:
@OndrejKotek Thanks for review! I will update the PR soon. This module is needed for exposing the `org.wildfly.extension.elytron.JaasCustomSecurityRealmWrapper` class so it can be referenced in the subsystem configuration

Reply:
This is because some class name has to be defined for the custom realm. I was thinking previously about adding a configuration parameter that would specify to use the wrapper like so:

```
/subsystem=elytron/custom-realm=myRealm:add(module=?,class-name=?,configuration={jaas-realm=true,entry=Entry1,module=jaasLoginModule,callback-handler=org.example.ExampleCallbackHandler,path=/my/example/path,relative-to=jboss.server.config.dir})
```

but then what would be defined under the module and class-name